### ISSUE TITLE
All custom fields return error: " deprecated invalid keys: customFields[myKey]. In future releases we'll throw exceptions on keys we do not recognize."

### ISSUE THREAD

**Issue by taylorstine** [braintree/braintree_node#88]
All custom fields I use throw this deprecation warning when calling `gateway.transaction.sale`.  I have all of the custom fields correctly registered.  It looks like [this is a bug](https://github.com/braintree/braintree_node/blob/master/src/braintree/transaction_gateway.coffee#L113), because this is causing the valid keys to be static, which might not be true for custom fields.

**Comment by crookedneighbor** [comment]
I think you're right that custom fields should not log that deprecation warning. 

One point of clarification, when you say `throw this deprecation warning`, you just mean that it is logging a warning, right? It's not actually throwing an error and preventing your code from running, correct?

**Comment by taylorstine** [comment]
Yes that's correct, it is just giving me a warning, not throwing an error.

**Comment by danakatz** [comment]
Thanks for pointing this out, @taylorstine! You'll see a fix for this in the next release - I'll update here when it's available.

**Comment by Amnacog** [comment]
I add there my issue because it belong to the same util function,
passing this object in transaction.sale
```
options: {
    submitForSettlement: true,
     threeDSecure: {
        required: true
    }
}
```


the resulted flattened keys in the verification util is:

```
[
    'options[submitForSettlement]',
    'threeDSecure[required]'
]
```

which provokes the deprecation warning because threeDSecure is not supposed to be in the parent.
Strange !
Hope I have helped !

**Comment by danakatz** [comment]
Hi @Amnacog, the forthcoming release should also resolve this and any other unnecessary deprecation warnings. Thanks for the additional info!

**Comment by danakatz** [comment]
This issue should now be fixed in 1.43.0, released today. Thanks again for letting us know about it! Feel free to post here if you see any other unexpected warnings.
